docs/csharp/language-reference/operators/member-access-operators.md

```diff
@@ -0,0 +1,163 @@
+
+  [!code-csharp[qualified name](~/samples/snippets/csharp/language-reference/operators/MemberAccessOperators.cs#QualifiedName)]
+
+  Use a [`using` directive](../keywords/using-directive.md) to make the use of qualified names optional.
+
+- Use `.` to access [type members](../../programming-guide/classes-and-structs/index.md#members), static and non-static, as the following code shows:
+
+  [!code-csharp-interactive[type members](~/samples/snippets/csharp/language-reference/operators/MemberAccessOperators.cs#TypeMemberAccess)]
+
+You can also use `.` to access an [extension method](../../programming-guide/classes-and-structs/extension-methods.md).
+
+## Indexer operator []
+
+Square brackets, `[]`, are typically used for array, indexer, or pointer element access.
+
+### Array access
+
+The following example demonstrates how to access array elements:
+
+[!code-csharp-interactive[array access](~/samples/snippets/csharp/language-reference/operators/MemberAccessOperators.cs#Arrays)]
+
+If an array index is outside the bounds of the corresponding dimension of an array, an <xref:System.IndexOutOfRangeException> is thrown.
+
+As the preceding example shows, you also use square brackets when you declare an array type or instantiate an array instance.
+
+For more information about arrays, see [Arrays](../../programming-guide/arrays/index.md).
+
+### Indexer access
+
+The following example uses .NET <xref:System.Collections.Generic.Dictionary%602> type to demonstrate indexer access:
+
+[!code-csharp-interactive[indexer access](~/samples/snippets/csharp/language-reference/operators/MemberAccessOperators.cs#Indexers)]
+
+Indexers allow you to index instances of a user-defined type in the similar way as array indexing. Unlike array indices, which must be integer, the indexer arguments can be declared to be of any type.
+
+For more information about indexers, see [Indexers](../../programming-guide/indexers/index.md).
+
+### Other usages of []
+
+For information about pointer element access, see [How to: access an array element with a pointer](../../programming-guide/unsafe-code-pointers/how-to-access-an-array-element-with-a-pointer.md).
+
+You also use square brackets to specify [attributes](../../programming-guide/concepts/attributes/index.md):
+
+```csharp
+[System.Diagnostics.Conditional("DEBUG")]
+void TraceMethod() {}
+```
+
+## Null-conditional operators ?. and ?[]
+
+Available in C# 6 and later, a null-conditional operator applies a member access, `?.`, or element access, `?[]`, operation to its operand only if that operand evaluates to non-null. If the operand evaluates to `null`, the result of applying the operator is `null`.
+
+The null-conditional operators are short-circuiting. That is, if one operation in a chain of conditional member or element access operations returns `null`, the rest of the chain doesn't execute. In the following example, `E` doesn't execute if `A`, `B`, or `C` evaluates to `null`:
```

Comment:
@BillWagner agree with you; I've made the chain shorter and updated the wording. Please take a look at the updated text.

Reply:
I think it is worth adding that if `B` is `null`, `C` is not evaluated. (as an example of the short circuiting).